File: sdkwork-backend-sdk-python/sdkwork_backend_sdk/api/channel_proxy.py
from typing import Any, Dict, List, Optional
from ..http_client import HttpClient
from ..models import PlusApiResultBoolean, PlusApiResultListPlusChannelProxyVO, PlusApiResultPagePlusChannelProxyVO, PlusApiResultPlusChannelProxyVO, PlusChannelProxyForm, QueryListForm

class ChannelProxyApi:
    """channel_proxy API client."""
    
    def __init__(self, client: HttpClient):
        self._client = client

    def update(self, body: PlusChannelProxyForm) -> PlusApiResultPlusChannelProxyVO:
        """Update channel proxy"""
        return self._client.put(f"/backend/v3/api/channel/proxy", json=body)

    def create(self, body: PlusChannelProxyForm) -> PlusApiResultPlusChannelProxyVO:
        """Create channel proxy"""
        return self._client.post(f"/backend/v3/api/channel/proxy", json=body)

    def list_by_page(self, body: Optional[QueryListForm] = None, params: Optional[Dict[str, Any]] = None) -> PlusApiResultPagePlusChannelProxyVO:
        """List channel proxies by page"""
        return self._client.post(f"/backend/v3/api/channel/proxy/list", json=body, params=params)

    def list_all_entities(self, body: Optional[QueryListForm] = None) -> PlusApiResultListPlusChannelProxyVO:
        """List all channel proxies"""
        return self._client.post(f"/backend/v3/api/channel/proxy/list/all", json=body)

    def get_by_id(self, id: str) -> PlusApiResultPlusChannelProxyVO:
        """Get channel proxy by ID"""
        return self._client.get(f"/backend/v3/api/channel/proxy/{id}")

    def delete(self, id: str) -> PlusApiResultBoolean:
        """Delete channel proxy"""
        return self._client.delete(f"/backend/v3/api/channel/proxy/{id}")

File: sdkwork-backend-sdk-flutter/lib/src/api/channel_proxy.dart
import '../http/client.dart';
import '../models.dart';

class ChannelProxyApi {
  final HttpClient _client;
  
  ChannelProxyApi(this._client);

  /// Update channel proxy
  Future<PlusApiResultPlusChannelProxyVO?> update(PlusChannelProxyForm body) async {
    final response = await _client.put(ApiPaths.backendPath('/channel/proxy'), body: body, contentType: 'application/json');
    return response is PlusApiResultPlusChannelProxyVO ? response : null;
  }

  /// Create channel proxy
  Future<PlusApiResultPlusChannelProxyVO?> create(PlusChannelProxyForm body) async {
    final response = await _client.post(ApiPaths.backendPath('/channel/proxy'), body: body, contentType: 'application/json');
    return response is PlusApiResultPlusChannelProxyVO ? response : null;
  }

  /// List channel proxies by page
  Future<PlusApiResultPagePlusChannelProxyVO?> listByPage(QueryListForm? body, Map<String, dynamic>? params) async {
    final response = await _client.post(ApiPaths.backendPath('/channel/proxy/list'), body: body, params: params, contentType: 'application/json');
    return response is PlusApiResultPagePlusChannelProxyVO ? response : null;
  }

  /// List all channel proxies
  Future<PlusApiResultListPlusChannelProxyVO?> listAllEntities(QueryListForm? body) async {
    final response = await _client.post(ApiPaths.backendPath('/channel/proxy/list/all'), body: body, contentType: 'application/json');
    return response is PlusApiResultListPlusChannelProxyVO ? response : null;
  }

  /// Get channel proxy by ID
  Future<PlusApiResultPlusChannelProxyVO?> getById(String id) async {
    final response = await _client.get(ApiPaths.backendPath('/channel/proxy/${id}'));
    return response is PlusApiResultPlusChannelProxyVO ? response : null;
  }

  /// Delete channel proxy
  Future<PlusApiResultBoolean?> delete(String id) async {
    final response = await _client.delete(ApiPaths.backendPath('/channel/proxy/${id}'));
    return response is PlusApiResultBoolean ? response : null;
  }
}
